scripts/main.py

```diff
@@ -529,25 +531,27 @@ def main():
             #############################################################
             manager = mp.Manager()
             res_queue = manager.Queue()
-            # Create a keyword argument dictionary to pass into the
-            # multiprocessor
-            keyword_args = {"infer": args.mode == "denoise",
-                            "intervals_file": infer_intervals,
-                            "exp_dir": args.exp_dir,
-                            "task": args.task,
-                            "peaks": args.peaks,
-                            "tracks": args.tracks,
-                            "num_workers": args.num_workers,
-                            "infer_threshold": args.threshold,
-                            "reg_rounding": args.reg_rounding,
-                            "batches_per_worker": args.batches_per_worker,
-                            "gen_bigwig": args.gen_bigwig,
-                            "sizes_file": args.genome,
-                            "res_queue": res_queue, "prefix": prefix,
-                            "deletebg": args.deletebg,
-                            "out_resolution": args.out_resolution}
-            write_proc = mp.Process(target=writer, kwargs=keyword_args)
-            write_proc.start()
+
+            if args.mode == "denoise":
```

Comment:
what does this condition catch for?

Reply:
Writer is only to be called for denoise mode. It's not relevant for eval, there's nothing being written to the `res_queue`. Not sure why we pass `res_queue` to eval_worker at all.